src/Microsoft.AspNetCore.Sockets.Client.Http/HttpConnection.cs

```diff
@@ -37,34 +38,46 @@ public class Connection : IConnection
         public event Action<byte[]> Received;
         public event Action<Exception> Closed;
 
-        public Connection(Uri url)
-            : this(url, null)
+        public HttpConnection(Uri url)
+            : this(url, TransportType.WebSockets)
         { }
 
-        public Connection(Uri url, ILoggerFactory loggerFactory)
+        public HttpConnection(Uri url, TransportType transportType)
+                    : this(url, transportType, loggerFactory: null)
+        {
+        }
+
+        public HttpConnection(Uri url, ILoggerFactory loggerFactory)
+            : this(url, TransportType.WebSockets, loggerFactory, httpMessageHandler: null)
+        {
+        }
+
+        public HttpConnection(Uri url, TransportType transportType, ILoggerFactory loggerFactory)
+            : this(url, transportType, loggerFactory, httpMessageHandler: null)
+        {
+        }
+
+        public HttpConnection(Uri url, TransportType transportType, ILoggerFactory loggerFactory, HttpMessageHandler httpMessageHandler)
         {
             Url = url ?? throw new ArgumentNullException(nameof(url));
 
             _loggerFactory = loggerFactory ?? NullLoggerFactory.Instance;
-            _logger = _loggerFactory.CreateLogger<Connection>();
+            _logger = _loggerFactory.CreateLogger<HttpConnection>();
```

Comment:
The HttpMessageHandler makes it a PITA, since it needs to be passed to the `DefaultTransportFactory`